test/LazyJsonParserTest.cpp

```diff
@@ -84,24 +84,9 @@ TEST(parseTest, parse) {
                 absl::StrCat(R"({"results": {"bindings": [)", result4b)},
                arrayPath);
 
-  // CHECK 4: Throw when exceeding input size limit.
-  // parseLiteral
-  EXPECT_ANY_THROW(LazyJsonParser::parse(
-      absl::StrCat(R"({"key":")", std::string(1'000'000, '0'), R"(")"), {}));
-  // BeforeArrayPath
-  EXPECT_ANY_THROW(LazyJsonParser::parse(
-      absl::StrCat(R"({"key":)", std::string(1'000'000, 0), R"(})"),
-      arrayPath));
-  // InArrayPath
-  EXPECT_ANY_THROW(LazyJsonParser::parse(
-      absl::StrCat(R"({"head":[)", std::string(1'000'000, 0)), {"head"}));
-  // AfterArrayPath
-  EXPECT_ANY_THROW(
-      LazyJsonParser::parse(absl::StrCat(R"({"head":[],"key": )",
-                                         std::string(1'000'000, '0'), R"(})"),
-                            {"head"}));
-
   // CHECK 5: Cornercases
-  EXPECT_NO_THROW(LazyJsonParser::parse("[1,2,3]", arrayPath));
-  EXPECT_NO_THROW(LazyJsonParser::parse("{}", arrayPath));
+  EXPECT_NO_THROW(expectYields("[1,2,3]", {}, arrayPath));
+  EXPECT_NO_THROW(expectYields("{}", {}, arrayPath));
+
+  EXPECT_ANY_THROW(expectYields(std::string(1'000'000, '0'), {}, arrayPath));
```

Comment:
The yellow branches inside the ` Ad_CORRECTNESS_CHECKs` seem to indicate that you haven't tested the case, that
e.g. there is a literal that is between a chunk.
But then again this looks odd to me, because you yield single characters.
Maybe we can both have a look at this.